File: soolicious/src/main/java/com/group5/soolicious/orders/Order.java
package com.group5.soolicious.orders;

import java.util.Date;
import java.util.List;

import com.group5.soolicious.customers.Customer;
import com.group5.soolicious.employees.Employee;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Order {
    private Integer id;

    private Customer customer;

    private Employee employee;

    private Date orderDate;

    private Double orderTotal;

    private String debitOrCredit;

    private String cardNumber;

    private String cvv;

    private String expiryDate;

    private String pickupMethod;

    private String timeToPickupOrDeliver;

    private String specialInstructions;

    private String lastUpdated;
    private List<OrderItem> orderItems;
}

File: soolicious-angular/src/app/shared/models/order.ts
import { Customer } from "./customer";
import { Dessert } from "./dessert";
import { Employee } from "./employee";

export interface Order {
    id?: number;
    customer?: Customer;
    employee?: Employee;
    orderDate?: Date;
    orderTotal?: number;
    debitOrCredit?: string;
    cardNumber?: string;
    cvv?: string;
    expiryDate?: string;
    expiryMonth?: string;  // Add this line
  expiryYear?: string;
    pickupMethod?: string;
    timeToPickupOrDeliver?: string;
    specialInstructions?: string;
    lastUpdated?: string;
    orderItems?: OrderItem[];
}

export interface OrderItem {
    dessert?: Dessert;
    quantityOrdered?: number;
    price?: number;
}
